### PR TITLE
[XAM/UI] Implemented profile passcode UI

### PR DESCRIPTION
**Pull request by AdrianCassar**

### CODE REVIEW

**Review by Tiger-Fury — APPROVED** [review]

File: src/xenia/kernel/xam/user_profile.h [hunk 1]
```diff
@@ -180,6 +180,9 @@ class UserProfile {
   uint32_t GetSubscriptionTier() const {
     return account_info_.GetSubscriptionTier();
   }
+  void GetPasscode(uint16_t* passcode) const {
+    memcpy(&passcode, account_info_.passcode, sizeof(account_info_.passcode));
```
Comment (Gliniak) [inline src/xenia/kernel/xam/user_profile.h]: &passcode That's a bug. It writes it to reference of a pointer. It should be without &

Comment (Gliniak) [inline src/xenia/kernel/xam/user_profile.h]: missing std::

File: src/xenia/kernel/xam/xam_ui.cc [hunk 2]
```diff
@@ -320,6 +320,158 @@ class MessageBoxDialog : public XamDialog {
...
 
+class ProfilePasscodeDialog : public XamDialog {
+ public:
+  ProfilePasscodeDialog(xe::ui::ImGuiDrawer* imgui_drawer, std::string& title,
+                        std::string& description, MESSAGEBOX_RESULT* result_ptr)
+      : XamDialog(imgui_drawer),
+        title_(title),
+        description_(description),
+        result_ptr_(result_ptr) {
+    memset(result_ptr, 0, sizeof(MESSAGEBOX_RESULT));
```
Comment (Gliniak) [inline src/xenia/kernel/xam/xam_ui.cc]: std::

File: src/xenia/kernel/xam/xam_ui.cc [hunk 3]
```diff
@@ -320,6 +320,158 @@ class MessageBoxDialog : public XamDialog {
...
+
+  bool SelectedSignedIn() const { return selected_signed_in_; }
+
+ private:
+  bool has_opened_ = false;
+  bool selected_signed_in_ = false;
+  std::string title_;
+  std::string description_;
+  const char* keys_[13] = {"None", "X",  "Y",    "RB",   "LB",   "LT",
+                           "RT",   "Up", "Down", "Left", "Right"};
```
Comment (Gliniak) [inline src/xenia/kernel/xam/xam_ui.cc]: Probably keys_ and keys_map can be yeeted outside and make it const

**Review by Gliniak — CHANGES_REQUESTED** [review]

**Pull request merged**